src/ctap2/mod.rs

```diff
@@ -809,5 +813,497 @@ pub(crate) fn set_or_change_pin_helper(
+                            BioEnrollmentResult::FingerprintSensorInfo(FingerprintSensorInfo {
+                                fingerprint_kind,
+                                max_capture_samples_required_for_enroll,
+                                max_template_friendly_name,
+                            }),
+                        )));
+                        return true;
+                    }
+                };
+            }
+            Err(e) => {
+                handle_errors!(e, status, callback, pin_uv_auth_result, skip_uv, skip_puap);
+            }
+        }
+    }
+    false
+}
+
+pub(crate) fn credential_management(
+    dev: &mut Device,
+    command: CredManagementCmd,
+    status: Sender<crate::StatusUpdate>,
+    callback: StateCallback<crate::Result<crate::ManageResult>>,
+    alive: &dyn Fn() -> bool,
+) -> bool {
+    let mut skip_uv = false;
+    let authinfo = match dev.get_authenticator_info() {
+        Some(i) => i.clone(),
+        None => {
+            callback.call(Err(HIDError::DeviceNotInitialized.into()));
+            return false;
+        }
+    };
+
+    if authinfo.options.cred_mgmt != Some(true)
+        && authinfo.options.credential_mgmt_preview != Some(true)
+    {
+        callback.call(Err(AuthenticatorError::HIDError(
+            HIDError::UnsupportedCommand,
+        )));
+        return false;
+    }
+
+    let use_legacy_preview = authinfo.options.cred_mgmt != Some(true);
+
+    let mut cred_management = match command {
+        CredManagementCmd::GetCredentials => {
+            CredentialManagement::new(CredManagementCommand::GetCredsMetadata, use_legacy_preview)
+        }
+        CredManagementCmd::DeleteCredential(cred_id) => CredentialManagement::new(
+            CredManagementCommand::DeleteCredential(cred_id),
+            use_legacy_preview,
+        ),
+        CredManagementCmd::UpdateUserInformation((cred_id, user)) => CredentialManagement::new(
+            CredManagementCommand::UpdateUserInformation((cred_id, user)),
+            use_legacy_preview,
+        ),
+    };
+
+    let mut credential_result = CredentialList::new();
+    let mut remaining_rps = 0;
+    let mut remaining_cred_ids = 0;
+    let mut current_rp = 0;
+    let mut skip_puap = false;
+    let mut pin_uv_auth_result = PinUvAuthResult::NoAuthRequired;
+    let mut pin = None;
+    while alive() {
+        if !skip_puap {
+            pin_uv_auth_result = match determine_puap_if_needed(
+                &mut cred_management,
+                dev,
+                skip_uv,
+                PinUvAuthTokenPermission::CredentialManagement,
+                UserVerificationRequirement::Preferred,
+                &status,
+                &callback,
+                alive,
+                &mut pin,
+            ) {
+                Ok(r) => r,
+                Err(()) => {
+                    return false;
+                }
+            };
+        }
+
```

Comment:
Having obtained a `pinUvAuthToken`, we should now send a status update containing a `Sender<CredManagementCommand>` and we should then poll the receiver.